TCMSApp/src/server/model/mschemas/user.js

```diff
@@ -1,48 +1,32 @@
-// import the necessary modules
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
+var validator = require('validator');
 
-// create  the user model
-
-// define schema
 var userSchema;
 userSchema = new Schema({
-    firstName: {type: String, required: true},
-    lastName: {type: String, required: true},
-    userName: {type: String, required: true},
-    password: {type: String, required: true},
+    firstName: {type: String},
+    lastName: {type: String},
+    fullName: {type: String},
+    password: {type: String},
+    currentProjectID: String,
+    trelloUserID: String,
     email: {
         type: String,
-        required: true,
         lowercase: true,
         validate: {
-            //TODO: make better regular expression for email. consider https://github.com/chriso/validator.js
             validator: function (v) {
-                return /^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i.test(v); // jshint ignore:line
```

Comment:
it seems that you've forgotten to merge with master. we already have this code there :)